src/projects/dualBackedYieldToOne/MDualBackedYieldToOne.sol

```diff
@@ -0,0 +1,320 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity 0.8.26;
+
+import { console } from "forge-std/console.sol";
+
+import { IERC20 } from "../../../lib/common/src/interfaces/IERC20.sol";
+
+import { MYieldToOne } from "../yieldToOne/MYieldToOne.sol";
+
+import { IMYieldToOne } from "../yieldToOne/IMYieldToOne.sol";
+
+import { IMDualBackedYieldToOne } from "./IMDualBackedYieldToOne.sol";
+
+import { ISwapFacility } from "../../swap/interfaces/ISwapFacility.sol";
+
+import { IMTokenLike } from "../../interfaces/IMTokenLike.sol";
+
+abstract contract MDualBackedYieldToOneStorageLayout {
+    struct MDualBackedYieldToOneStorageStruct {
+        address secondaryToken;
```

Comment:
we can safely assume token complies with IERC20Metadata.sol standard, no need to save decimals